inc/features/blocks/personalization/components/goal-picker.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+
+const { SelectControl } = wp.components;
+const { __ } = wp.i18n;
+
+const registeredGoals = window.Altis.Analytics.Experiments.Goals || {};
+
+const GoalPicker = ( { goal, onChange } ) => {
+
+	const goals = Object.entries( registeredGoals );
+
+	if ( goals.length < 1 ) {
+		return null;
+	}
+
+	const options = [
+		{
+			label: __( 'Impressions', 'altis-experiments' ),
+			value: '',
+		},
+	];
+
+	// Add registered goals to dropdown.
+	goals.forEach( ( [ name, data ] ) => {
+		options.push( {
+			label: data.label || name,
+			value: name,
+		} );
+	} );
```

Comment:
Rather than mutating, could we map?

```suggestion
	const options = [
		{
			label: __( 'Impressions', 'altis-experiments' ),
			value: '',
		},
		...goals.map( ( [ name, data ] ) => ( {
			label: data.label || name,
			value: name,
		} ),
	];
```

Reply:
We could but feels like splitting hairs. Mutation isn't inherently bad, moreso in specific circumstances.